.github/drafts/17478/17478_Okta_Draft.md

```diff
@@ -0,0 +1,114 @@
+# Okta
+
+Okta is an Identity-as-a-Service (IDaaS) provider. To configure OpenID Connect (OIDC) within Appsmith using Okta, follow these steps:
+
+{% hint style="info" %}
+OpenID Connect is available **only in the** [**enterprise edition**](https://www.appsmith.com/pricing) for **self-hosted instances**, and only the **Superuser** of your **Appsmith Instance** can set up **OIDC**.
+{% endhint %}
+
+### **Create Application**
+
+* Log in to your [Okta Developer Account](https://developer.okta.com) and go to **Applications**. (Please create an account if you don’t have one).
+
+![Create a new app on Okta for SSO Configuration](../../../../../.gitbook/assets/Okta-Create-Application.png)
+
+* Click on **Create App integration**. In Create App integration window, choose **OIDC** as the Sign-in method and **Web application** as the Application type. Click on Next.
+
+![New App Configurations](../../../../../.gitbook/assets/Okta-Create-App-Integration-Config.png)
+
+* In the **General settings** of the newly created App Integration:
+  * Give your App integration a meaningful name (Optional.)
+
+![New App - General Settings](./as_Okta_edited.png) <!-- replaced for 17478 -->
+
+* Under the **Grant Type > Client acting on behalf of a user** heading, tick the **Refresh Token** checkbox (Optional - Highly recommended).
+
+* Add the redirect URL (Copied from the [OIDC window in Appsmith’s Admin Settings](./#capture-redirect-url-for-sso-configuration)) in the **Sign-in redirect URL** field.
+
+![Redirect URL available at Appsmith >> Profile >> Admin Settings >> Authentication >> OIDC](../../../../../.gitbook/assets/Appsmith-Admin-Settings-Authentication-OIDC-RedirectURL.png)
+
+You can add multiple redirect URLs.
+
+![Okta - Sign-in redirect URIs](../../../../../.gitbook/assets/Okta-Sign-in-redirect-URis.png)
+
+* Under Assignment, configure the Accessibility of this App as required.
+
+![Configure Assignments](../../../../../.gitbook/assets/Okta-Assignments.png)
+
+* Click on **Save**. This action creates the app integration and opens the settings page to configure additional options.
+
+### Configure Okta fields in Appsmith
+
+To continue with the OIDC setup on Appsmith, navigate to the fields on the Okta configurations, and perform the actions as mentioned below:
+
+*   On the Okta Application homepage, go to the **General** tab:
+
+    * Copy the **Client ID** and **Client Secret** and paste them into the **OIDC** configurations in Appsmith.
+
+
+
+![Client ID, Client Secret, and Okta Domain](<../../../../../.gitbook/assets/Screenshot 2022-07-18 at 11.45.21 AM.png>)
```

Comment:
Can we please rename the image.

Reply:
Updated the image caption to: "Use these Okta values to configure OIDC on Appsmith"

Let me know if you have a different suggestion!